open-notes/maturity-model.md

```diff
@@ -0,0 +1,394 @@
+
+### Regular team retrospective
+
+The team has a regular moment, AT MINIMUM monthly, to reflect on how things have gone over the last while and suggest improvements, which they then implement. Hopefully one of the first things to come out is monthly is too long a time to reflect on possible improvements.
+
+### Centralized Structured Logging
+
+Simply logging is not enough. Logs should be collected in a central place where they can be easily searched. To facilitate the searching, logs should be structured, rather than free text logs.
+
+### No secrets in source control
+
+No secrets to any environments should be in source control. Not just the current HEAD but any previous secrets should be scrubbed from history with something like [BFG](https://rtyley.github.io/bfg-repo-cleaner/).
+
+### Deploy monthly
+
+The minimum is that each application is deployed at least once per month. This includes applications that have not changed in that month to ensure that pipelines are being maintained.
+
+### Database backups
+
+Databases have a backup schedule. Databases are not considered backed up unless the backups are tested regularly. There is a process documented for restore.
+
+### 4 eyes principle
+
+Pull Requests are only merged when at least one other person has reviewed and agreed that the PR is ok.
+
+### Basic health endpoint
+
+All services have a basic `health` endpoint returning back a `200OK` when the service is running.
+
+### Refinement
+
+It is important that stories are refined and that those at the top of the list are in a state where they can be picked up. This skill asserts that the team has a process and techniques that make sure that this is always the case. Typically, this is only achieved if stories are worked on collaboratively with people with different skill and knowledge sets.  
+
+Some recommended techniques for knowledge mining:
+
+- 3 Amigos: Divide and conquer so the team can refine stories in smaller groups. This is often more effective and so any final refinement is done with stories fleshed out from multiple points of view.
+- Example mapping: A great way to flesh out stories and give clear examples. Great for identifying edge-cases and expected behaviour. Also easy to do collaboratively.
+- EventStorming: A great technique for knowledge mining. It makes clear current process to all involved and makes it easy to talk about needed changes or dive into new behaviour.
+
+### Infrastructure as Code
+
+All infrastructure should be created via selected code provider such as Terraform.
+
+### Validated configuration
+
+Configuration should be validated, at latest, at application startup. If you use conventions, it is useful to place the environment in the names so they can be validated for the current environment. For this reason it is best to have them at the start or end of the name.
+### Resources are tagged
```

Comment:
Tag resources is a good way to starts. For more evolved environments the concept of FinOps is important, and one implementation is cost centres, which are attached nicely with the finance domain.